webapi/CopilotChat/Skills/ChatSkills/ChatSkill.cs

```diff
@@ -313,109 +294,112 @@ public async Task<string> ExtractAudienceAsync(SKContext context)
     /// <summary>
     /// Generate the necessary chat context to create a prompt then invoke the model to get a response.
     /// </summary>
+    /// <param name="chatId">The chat ID</param>
+    /// <param name="userId">The user ID</param>
     /// <param name="chatContext">The SKContext.</param>
-    /// <returns>A response from the model.</returns>
-    private async Task<string> GetChatResponseAsync(string chatId, SKContext chatContext)
+    /// <returns>The created chat message.</returns>
+    private async Task<ChatMessage?> GetChatResponseAsync(string chatId, string userId, SKContext chatContext)
     {
-        // 0. Get the audience
-        await this.UpdateResponseStatusOnClient(chatId, "Extracting audience");
-        var audience = await this.GetAudienceAsync(chatContext);
-        if (chatContext.ErrorOccurred)
-        {
-            return string.Empty;
-        }
-
-        // 1. Extract user intent from the conversation history.
-        await this.UpdateResponseStatusOnClient(chatId, "Extracting user intent");
-        var userIntent = await this.GetUserIntentAsync(chatContext);
+        // Get the audience and user intent
+        await this.UpdateBotResponseStatusOnClient(chatId, "Extracting audience and user intent");
+        var tasks = await Task.WhenAll(this.GetAudienceAsync(chatContext), this.GetUserIntentAsync(chatContext));
```

Comment:
Is this going to create a data race with the same chatContext? i.e one of the method results in error but the other one finishes later and override the error state.

Reply:
reverted. the audience extraction code will no longer be async after #19